backend/app/services/chatbot_service.py

```
@@ -23,53 +27,55 @@ async def validate_user(self, user_id: ObjectId):
-            chatbot.products = products
-            chatbot.user_id = user_id
-            
-            logger.debug(f"Saving chatbot to database")
+
+            # Generate a unique file name using user_id, current timestamp, and a unique suffix.
+            timestamp = datetime.utcnow().strftime("%Y%m%d%H%M%S")
+            unique_suffix = uuid.uuid4().hex[:6]  # A short random string.
+            products_file_name = f"products_user_{str(user_id)}_{timestamp}_{unique_suffix}.json"
+
+            try:
+                with open(products_file_name, "w", encoding="utf-8") as f:
```

Ruff (0.8.2)

Async functions should not open files with blocking methods like `open`

(ASYNC230)